source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -195,7 +195,9 @@ class Filter : public Logger::Loggable<Logger::Id::ext_proc>,
 
   void sendBufferedData(ProcessorState& state, ProcessorState::CallbackState new_state,
                         bool end_stream) {
-    sendBodyChunk(state, *state.bufferedData(), new_state, end_stream);
+    if (state.hasBufferedData()) {
+      sendBodyChunk(state, *state.bufferedData(), new_state, end_stream);
```

Comment:
Another observation: the call in case of an empty body, will not change the state to `new_state`. Is this the desired outcome?

Reply:
Yeah, that's a good catch.  We should send an empty body if no buffered data as in the new diffs.